### PR TITLE
Add unit test and test data for functions in handle_files.py

### PR DESCRIPTION
**Pull request by mzfr**
This contains unit tests for all the function in `handle_files.py` except for `addon_file_exists` function. I tried but couldn't write successful test for that function.

### CODE REVIEW

File: tests/test_handle_files.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import unittest
+from os.path import abspath, dirname, join
+
+from kodi_addon_checker.common import relative_path
```
Comment (Rechi) [inline tests/test_handle_files.py]: unused

File: tests/test_handle_files.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from kodi_addon_checker.handle_files import create_file_index
+from kodi_addon_checker.handle_files import find_file_recursive as FFR
+
+HERE = abspath(dirname(__file__))
+
+
+class TestFindFile(unittest.TestCase):
+
+    def setUp(self):
+        self.path = "/tmp"
```
Comment (Rechi) [inline tests/test_handle_files.py]: `self.path = tempfile.mkdtemp()`, directory has to be deleted explicitly at the end of the test

File: tests/test_handle_files.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        self.file = open("/tmp/addon.txt", "w+")
```
Comment (Rechi) [inline tests/test_handle_files.py]: `os.path.join(self.path, "addon.txt")`

File: tests/test_handle_files.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        self.assertIsNone(find_file("randomname", self.path))
+
+    def test_find_file_is_not_none(self):
+        self.assertIsNotNone(find_file("addon.txt", self.path))
+
+
+class TestFindFileRecursive(unittest.TestCase):
+
+    def setUp(self):
+        self.path = "/tmp"
```
Comment (Rechi) [inline tests/test_handle_files.py]: `self.path = tempfile.mkdtemp()`

File: tests/test_handle_files.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        self.file = open("/tmp/addon.txt", "w+")
```
Comment (Rechi) [inline tests/test_handle_files.py]: `os.path.join(self.path, "addon.txt")`

File: tests/test_handle_files.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    def setUp(self):
+        self.path = tempfile.mkdtemp()
+        self.file = join(self.path, "addon.txt")
+
+    def test_find_file_is_none(self):
+        self.assertIsNone(find_file("randomname", self.path))
+
+    def test_find_file_is_not_none(self):
+        with open(self.file, "w+"):
```
Comment (Razzeee) [inline tests/test_handle_files.py]: Why does this need this line? Why does the function above not need this?

File: tests/test_handle_files.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+class TestFindFileRecursive(unittest.TestCase):
+
+    def setUp(self):
+        self.path = tempfile.mkdtemp()
+        self.file = join(self.path, "addon.txt")
+
+    def test_find_file_is_none(self):
+        self.assertIsNone(FFR("randomname", self.path))
+
+    def test_find_file_is_not_none(self):
```
Comment (Razzeee) [inline tests/test_handle_files.py]: This test is fine, but I would like to also have a test for a file being deeper in a directory structure and still being correctly found by this.

File: tests/test_data/addon.xml [hunk 8]
```diff
@@ -0,0 +1,2 @@
+<?xml version='1.0' encoding='UTF-8' standalone='yes'?>
```
Comment (Razzeee) [inline tests/test_data/addon.xml]: Why is this here? I don't think your using it.

File: tests/test_handle_files.py (hunk 6 above)
Comment (mzfr) [reply tests/test_handle_files.py]: This creates a new file on a given path so when tests are run the file can be found on a given location.

File: tests/test_handle_files.py [hunk 9]
```diff
@@ -0,0 +1,63 @@
+import unittest
+import tempfile
+import shutil
+from pathlib import Path
+from os import makedirs
+from os.path import abspath, dirname, join
+
+from kodi_addon_checker.handle_files import find_file
+from kodi_addon_checker.handle_files import create_file_index
+from kodi_addon_checker.handle_files import find_file_recursive as FFR
```
Comment (Rechi) [inline tests/test_handle_files.py]: `find_files_recursive`

File: tests/test_handle_files.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+    def test_find_file_is_not_none(self):
+        with open(self.file, "w+"):
+            self.assertIsNotNone(find_file("addon.txt", self.path))
+
+    def teardown(self):
+        shutil.rmtree(self.path)
+
+
+class TestFindFileRecursive(unittest.TestCase):
```
Comment (Rechi) [inline tests/test_handle_files.py]: `TestFindFilesRecursive`

File: tests/test_handle_files.py [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+    def setUp(self):
+        self.path = tempfile.mkdtemp()
+        self.file = join(self.path, "addon.txt")
+        self.dirs = join(self.path, "tree/subdir")
+        makedirs(self.dirs)
+        self.folder = join(self.dirs, "addon.xml")
+
+    def test_find_file_recursive_is_none(self):
```
Comment (Rechi) [inline tests/test_handle_files.py]: `test_find_files_recursive_is_none`

File: tests/test_handle_files.py [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+        self.assertFalse(list(FFR("randomname", self.path)))
+
+    def test_find_file_recursive_is_not_none(self):
```
Comment (Rechi) [inline tests/test_handle_files.py]: `test_find_files_recursive_is_not_none`

File: tests/test_handle_files.py [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+        with open(self.file, "w+"):
+            self.assertIsNotNone(FFR("addon.txt", self.path))
+
+    def test_find_file_recursive_deeply(self):
```
Comment (Rechi) [inline tests/test_handle_files.py]: `test_find_files_recursive_deeply`

File: tests/test_handle_files.py [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+class TestFindFilesRecursive(unittest.TestCase):
+
+    def setUp(self):
+        self.path = tempfile.mkdtemp()
+        self.file = join(self.path, "addon.txt")
+        self.dirs = join(self.path, "tree/subdir")
+        makedirs(self.dirs)
+        self.folder = join(self.dirs, "addon.xml")
+
+    def test_find_files_recursive_is_none(self):
```
Comment (Rechi) [inline tests/test_handle_files.py]: function calls `assertFalse`, so None isn't expected

File: tests/test_handle_files.py [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+        self.assertFalse(list(FFR("randomname", self.path)))
+
+    def test_find_files_recursive_is_not_none(self):
+        with open(self.file, "w+"):
+            self.assertIsNotNone(FFR("addon.txt", self.path))
```
Comment (Rechi) [inline tests/test_handle_files.py]: correct result should be checked

File: tests/test_handle_files.py [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+    def test_find_files_recursive_deeply(self):
+        with open(self.folder, "w+"):
+            self.assertIsNotNone(FFR("addon.xml", self.path))
```
Comment (Rechi) [inline tests/test_handle_files.py]: correct result should be checked

**Comment by mzfr** [comment]
@Razzeee anything needed to be done in these?
